cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -408,7 +410,8 @@ private RejectedExecutionHandler createAsyncRejectedExecutionHandler() {
       @Override
       public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
         if (!executor.isShutdown()) {
-          metricsCollector.increment("collect.async.reject", 1);
+          metricsCollector.childCollector(Constants.Metrics.Tag.NAMESPACE, Constants.SYSTEM_NAMESPACE)
```

Comment:
Thinking about how to cache it. We may be able to get by without requiring it now, but when streams can be in different namespaces, we'll probably have to call ``childCollector`` atleast once. I'll try to cache the rest of the calls for now.